binderhub/utils.py

```diff
@@ -1,8 +1,37 @@
 """Miscellaneous utilities"""
 from collections import OrderedDict
+from hashlib import blake2b
+from math import log
+
 from traitlets import Integer, TraitError
 
 
+def blake2b_as_float(b):
+    """Compute Blake2 digest and convert it to a float.
```

Comment:
Described that the input is bytes and renamed it to `blake2b_as_int`. An alternative name could be `blake2b_digest_as_int` but that feels a bit long :-/

Reply:
I was thinking more in lines of get_hash_as_int or similarly as a function name, as I figure it is irrelevant to know the hash implementation algorithm and it increase readability of the code reading get_hash_as_int as compared to get_blake2b_as_int for example as it would require the code-reader to read up on blake2b, and perhaps start wondering if it was important to use blake2b specifically, etc.

Reply:
I'd be hesitant to use the word `hash` because there is a builtin function in Python called `hash()` that returns integers. The problem with `hash()` is that we can not use it because it is a carefully constructed hash function for the use cases of CPython itself. It doesn't behave like a hash function where two similar inputs produce wildly different outputs.

For example `hash(i) == i` for most (small) integers and similarly `hash(-3) == -3` and other "weird" stuff. So it matters that we don't use `hash()` but it doesn't matter so much which of the hash functions from `hashlib` we use. Some of them might be more expensive to compute or produce results which aren't easy to turn into 64bit integers.

Reply:
For those that has this python insight could verify that that was not a concern perhaps. I like the word hash as it is what I've come to associate with the idea of taking something big and generating something small.

I felt the idea was valid as we would be fine to swap blake2b for some other hash function later if we felt like it and that would been fine, as the way we use the function remains - we want to get a hash from binary data as a number.

Reply:
I renamed it `blake2b_hash_as_int` as a way to help the user who doesn't know that Blake2 is a hash. I added a comment in `rendezvous_rank()` to explain some of the constraints on what hash function can be used here.

Because `blake2b_hash_as_int()` lives in a public namespace, to be used by who ever needs it, I think it should have a name that tells the caller what it does. In `rendezvous_rank()` we don't care so much so I added a comment explaining that.

Reply:
resolved with "blake2b hash as int"